sdk/go/x/auto/stack.go

```diff
@@ -0,0 +1,772 @@
+	if err != nil {
+		return nil, newAutoError(errors.Wrap(err, "failed to get stack history"), stdout, stderr, errCode)
+	}
+
+	var history []UpdateSummary
+	err = json.Unmarshal([]byte(stdout), &history)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to unmarshal history result")
+	}
+
+	return history, nil
+}
+
+// GetConfig returns the config value associated with the specified key.
+func (s *Stack) GetConfig(ctx context.Context, key string) (ConfigValue, error) {
+	return s.Workspace().GetConfig(ctx, s.Name(), key)
+}
+
+// GetAllConfig returns the full config map.
+func (s *Stack) GetAllConfig(ctx context.Context) (ConfigMap, error) {
+	return s.Workspace().GetAllConfig(ctx, s.Name())
+}
+
+// SetConfig sets the specified config KVP.
+func (s *Stack) SetConfig(ctx context.Context, key string, val ConfigValue) error {
+	return s.Workspace().SetConfig(ctx, s.Name(), key, val)
+}
+
+// SetAllConfig sets all values in the provided config map.
+func (s *Stack) SetAllConfig(ctx context.Context, config ConfigMap) error {
+	return s.Workspace().SetAllConfig(ctx, s.Name(), config)
+}
+
+// RemoveConfig removes the specified config KVP.
+func (s *Stack) RemoveConfig(ctx context.Context, key string) error {
+	return s.Workspace().RemoveConfig(ctx, s.Name(), key)
+}
+
+// RemoveAllConfig removes all values in the provided list of keys.
+func (s *Stack) RemoveAllConfig(ctx context.Context, keys []string) error {
+	return s.Workspace().RemoveAllConfig(ctx, s.Name(), keys)
+}
+
+// RefreshConfig gets and sets the config map used with the last Update.
+func (s *Stack) RefreshConfig(ctx context.Context) (ConfigMap, error) {
+	return s.Workspace().RefreshConfig(ctx, s.Name())
+}
+
+// Info returns a summary of the Stack including its URL.
+func (s *Stack) Info(ctx context.Context) (StackSummary, error) {
+	var info StackSummary
+	err := s.Workspace().SelectStack(ctx, s.Name())
+	if err != nil {
+		return info, errors.Wrap(err, "failed to fetch stack info")
+	}
+
+	summary, err := s.Workspace().Stack(ctx)
+	if err != nil {
+		return info, errors.Wrap(err, "failed to fetch stack info")
+	}
+
+	if summary != nil {
+		info = *summary
+	}
+
+	return info, nil
+}
+
+// UpdateSummary provides an summary of a Stack lifecycle operation (up/preview/refresh/destroy).
+type UpdateSummary struct {
+	Kind        string            `json:"kind"`
+	StartTime   string            `json:"startTime"`
+	Message     string            `json:"message"`
+	Environment map[string]string `json:"environment"`
+	Config      ConfigMap         `json:"config"`
+	Result      string            `json:"result,omitempty"`
+
+	// These values are only present once the update finishes
+	EndTime         *string         `json:"endTime,omitempty"`
+	ResourceChanges *map[string]int `json:"resourceChanges,omitempty"`
+}
+
+// OutputValue models a Pulumi Stack output, providing the plaintext value and a market indicating secretness.
```

Comment:
I'm not sure what you meant by "a market indicating secretness", so dunno what to suggest here 😄

Reply:
😆  ah it's supposed to be marker ✏️